instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -263,3 +273,23 @@ def telemetry_interceptor(request_or_iterator, context):
         return _wrap_rpc_behavior(
             continuation(handler_call_details), telemetry_wrapper
         )
+
+    # Handle streaming responses separately - we have to do this
+    # to return a *new* generator or various upstream things
+    # get confused, or we'll lose the consistent trace
+    def _intercept_server_stream(
+        self, behavior, handler_call_details, request_or_iterator, context
+    ):
+
+        with self._set_remote_context(context):
+            with self._start_span(handler_call_details, context) as span:
+                context = _OpenTelemetryServicerContext(context, span)
+
+                try:
+                    yield from behavior(request_or_iterator, context)
+
+                except Exception as error:
+                    # pylint:disable=unidiomatic-typecheck
+                    if type(error) != Exception:
```

Comment:
Can you please explain why is this necessary? `record_exception` expects an `Exception`. Even so, if this check was appropriate, wouldn't it be better to use `isinstance`?

Reply:
Yes - in the gRPC world bare `Exception` instances are thrown by the system when you call `grpc.abort()`, and some of those shouldn't be treated as actual errors, for example the HTTP 404 series responses which are returned to the client but not recorded as errors in a trace.

This allows us to record anything *except those* as uncaught errors, since it's really unlikely anything else would throw a bare exception itself.  It's not ideal but there just isn't any other way to distinguish these at this point in the program flow.

Using `isinstance()` would catch application exceptions here, which we want to raise normally.

Reply:
Ok, I see. This is unusual code, but necessary because of a particularity of gRPC that the reader may not know about. I'll approve, but please add this explanation as a comment here. :+1:

Reply:
The same exact code is on line 263, with a comment describing why it was done this way.  Happy to duplicate the comment here if that would be preferred.